src/format/command.rs

```diff
@@ -33,14 +33,18 @@ fn check_formatting(formatted_files: Vec<Formatted>) -> Result<(), Error> {
     if problem_files.is_empty() {
         Ok(())
     } else {
-        Err(Error::Format { problem_files })
+        Err(Error::Format {
+            input: FormatInput::Files {
+                problem_files: problem_files,
+            },
+        })
     }
 }
 
 fn write_formatted(formatted_files: Vec<Formatted>) -> Result<(), Error> {
     for formatted in formatted_files {
         let path = formatted.path;
-        let mut f = File::open(&path).map_err(|e| Error::FileIO {
+        let mut f = File::create(&path).map_err(|e| Error::FileIO {
```

Comment:
What does this change do?

Reply:
This opens the file with write permissions. Using `File::open` I was getting permission errors when it tried to write the newly formatted file, since `File::open` only requests read permissions.

Reply:
Thank you